app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala

```diff
@@ -50,14 +52,16 @@ class HaulierStatusService @Inject()(haulierStatusRepository: HaulierStatusRepos
     EitherT.fromOptionF(haulierStatusRepository.findAndDelete(correlationId), CorrelationIdNotFound)
 
   def updateStatus()(implicit headerCarrier: HeaderCarrier): Future[Future[HttpResponse]] =
-    haulierStatusRepository.findAllOlderThan(appConfig.notificationCheckSchedulerIntervalSeconds).map {
-      case Nil =>
+    haulierStatusRepository.findAllOlderThan(appConfig.intervalSeconds, appConfig.limit).map { documents =>
+      val receivedDocPercentage = (documents.count(_.status == Received).toFloat / documents.size) * 100
+      if (documents.isEmpty || (receivedDocPercentage >= appConfig.threshold)) {
         logger.info("Setting haulier status to AVAILABLE")
         customsServiceStatusConnector.updateStatus(appConfig.haulierServiceId, AVAILABLE)
-      case correlationIds =>
+      } else {
+        val createdDocs = documents.filter(_.status == Created)
         logger.warn("Setting haulier status to UNAVAILABLE")
-        logger.info(s"${correlationIds.length} Correlation ids found (curtailed): ${correlationIds.takeRight(10).mkString(", ")}")
+        logger.info(s"${createdDocs.length} Correlation ids found (curtailed): ${createdDocs.map(_.id).takeRight(10).mkString(", ")}")
```

Comment:
this existing logging, can we do json pretty print? 

is it worth logging `receivedDocPercentage` ?

Reply:
I have changed the logging to log only the corrIds, think no need of logging full mongo document. 
I dont see any benefit of logging the receivedDocPercentage , but can do if you think its useful

Reply:
time also please if possible, currently is logs `HaulierStatusDocument(ab8f4536-80a8-4827-a930-638de299656d,2024-02-27T06:42:11.822Z)`, 
would be good if it is ab8f4536-80a8-4827-a930-638de299656d, 2024-02-27T06:42:11.822Z . gives an idea of when was it created/updated. 

I do think receivedDocPercentage is useful to know in UNAVAILABLE cases. if we keep on seeing a certain range and if there are no real issues, then we can set lower our threshold to match that. 

and it is worth to log even in AVAILABLE cases so we can see in future if we can increase our threshold from current 90%

Reply:
I think if we were going to log this level of detail, we would definitely want some of it to be debug (the percentage for instance), so that we can easily switch it off. It might have some value prior to first release to see the percentage, but after that, it will add a lot of noise.